pkg/api/v1/clients/multicluster/client_manager.go

```diff
@@ -0,0 +1,92 @@
+package multicluster
+
+import (
+	"context"
+	"sync"
+
+	"github.com/solo-io/go-utils/contextutils"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients"
+	"github.com/solo-io/solo-kit/pkg/multicluster/handler"
+	"go.uber.org/zap"
+	"k8s.io/client-go/rest"
+)
+
+//go:generate mockgen -destination=./mocks/client_manager.go -source client_manager.go -package mocks
+
+// Allows objects to register callbacks with a ClusterClientManager.
+type ClusterClientHandler interface {
+	HandleNewClusterClient(cluster string, client clients.ResourceClient)
+	HandleRemovedClusterClient(cluster string, client clients.ResourceClient)
+}
+
+// Stores clients for clusters as they are discovered by a config watcher.
+// Implementation in this package allows registration of ClusterClientHandlers.
+type ClusterClientManager interface {
+	handler.ClusterHandler
```

Comment:
i don't think `handler.ClusterHandler` is required for this interface, is it? i'm thinking it will be easier to mock this interface if we don't explicitly require Clusterhandler here